### PR TITLE
DROOLS-3219: [DMN Designer] Nodes can not be added once autolayout applied

### PR DESCRIPTION
**Pull request by danielzhe**

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/DMNDiagramEditor.java [hunk 1]
```diff
@@ -143,7 +148,15 @@ public void onStartup(final ObservablePath path,
 
     @Override
     public void open(final ProjectDiagram diagram) {
+        final Graph<?, Node<org.kie.workbench.common.stunner.core.graph.content.view.View, ?>> graph = diagram.getGraph();
         this.layoutHelper.applyLayout(diagram.getGraph());
+        final Optional<Node<org.kie.workbench.common.stunner.core.graph.content.view.View, ?>> dd = StreamSupport.stream(graph.nodes().spliterator(),
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/editor/DMNDiagramEditor.java]: I think putting this in `LayoutHelper.applyLayout(..)` would be much better (and avoid the need to copy and paste). The issue will not be limited to DMN (as the error had nothing specifically to do with DMN) but also affect _any_ diagram that invokes the "auto layout code" (so BPMN could be equally affected). 

The code can also be simplified:
```
graph.nodes().spliterator(), false)
    .filter(n -> n.getContent().getDefinition() instanceof DMNDiagram)
    .findFirst()
    .ifPresent(diagram -> diagram.getContent().setBounds(BoundsImpl.build(0, 0, 0, 0)));
```

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/SessionDiagramEditorScreen.java [hunk 2]
```diff
@@ -298,7 +304,15 @@ public void onError(final ClientRuntimeError error) {
     void open(final Diagram diagram,
               final Command callback) {
         screenPanelView.setWidget(presenter.getView());
-        layoutHelper.applyLayout(diagram.getGraph());
+        final Graph<?, Node<View, ?>> graph = diagram.getGraph();
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp/src/main/java/org/kie/workbench/common/dmn/showcase/client/screens/editor/SessionDiagramEditorScreen.java]: See my earlier comment about centralising the fix.

**Comment by danielzhe** [comment]
I don't even know that was a Jira about this... Changing now.

**Comment by danielzhe** [comment]
@manstis Changes done.

**Comment by karreiro** [comment]
Jenkins please retest this.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/components/layout/LayoutHelper.java [hunk 3]
```diff
@@ -35,10 +40,22 @@ public LayoutHelper(final LayoutService layoutService,
         this.layoutExecutor = layoutExecutor;
     }
 
-    public void applyLayout(final Graph graph) {
+    public void applyLayout(final Diagram diagram) {
+        final Graph graph = diagram.getGraph();
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/components/layout/LayoutHelper.java]: I think we should declare graph as `Graph<?, Node>` and then the implementation is slightly more readable:
```java
public void applyLayout(final Diagram diagram) {
        final Graph<?, Node> graph = diagram.getGraph();
        if (graph != null && !this.layoutService.hasLayoutInformation(graph)) {
            final Layout layout = this.layoutService.createLayout(graph);
            this.layoutExecutor.applyLayout(layout, graph);

            for (final Node node : graph.nodes()) {
                if (CanvasLayoutUtils.isCanvasRoot(diagram, node)) {
                    if (node.getContent() instanceof HasBounds) {
                        ((HasBounds) node.getContent()).setBounds(BoundsImpl.build(0, 0, 0, 0));
                    }
                }
            }
        }
    }
```

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/layout/LayoutHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    private static boolean isZero(final Bounds.Bound bound) {
+        return isCloseToZero(bound.getX())
+                && isCloseToZero(bound.getY());
+    }
+
+    private static boolean isCloseToZero(final double value) {
+        return Math.abs(value - 0) < 0.01;
+    }
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/components/layout/LayoutHelperTest.java]: JUnit already contains methods for equality comparison with some tolerance:
```java
@Test
    public void applyLayout() {

        final LayoutHelper helper = new LayoutHelper(layoutService, layoutExecutor);
        helper.applyLayout(diagram);
        verify(rootNodeContent).setBounds(argumentCaptor.capture());

        final Bounds bounds = argumentCaptor.getValue();

        isZero(bounds.getLowerRight());
        isZero(bounds.getUpperLeft());
    }

    private static void isZero(final Bounds.Bound bound) {
        assertEquals(0.0, bound.getX(), 0.01);
        assertEquals(0.0, bound.getY(), 0.01);
    }
```

**Pull request merged**